homeassistant/components/airly/__init__.py

```diff
@@ -11,27 +12,39 @@
 from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
 from homeassistant.core import Config, HomeAssistant
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
-from homeassistant.util import Throttle
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import (
     ATTR_API_ADVICE,
     ATTR_API_CAQI,
     ATTR_API_CAQI_DESCRIPTION,
     ATTR_API_CAQI_LEVEL,
-    DATA_CLIENT,
     DOMAIN,
+    MAX_REQUESTS_PER_DAY,
     NO_AIRLY_SENSORS,
 )
 
+PLATFORMS = ["air_quality", "sensor"]
+
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_SCAN_INTERVAL = timedelta(minutes=10)
+
+def update_interval(hass, unload=False):
+    """Return update_interval for configured Airly instances."""
+    # We check how many Airly config entries are and calculate interval to not
+    # exceed allowed numbers of requests.
+    instances = len(hass.config_entries.async_entries(DOMAIN))
+    if unload:
```

Comment:
Why not base it on `hass.data` and you can skip this ?

Reply:
Current logic is flawed too, because unloading doesn't mean deletion.

Reply:
Now the integration uses `hass.config_entries.async_entries(DOMAIN)` in `async_setup_entry` and `hass.data[DOMAIN]` in `async_unload_entry` to calculate the number of Airly instances. The `set_update_interval` method sets `update_interval` to all instances.